### ISSUE TITLE
Kitty's Keyboard Protocol is not deinitialized when both main and alternate screens are used

### ISSUE THREAD

**Issue by Mukundan314** [dankamongmen/notcurses#2509]
Leaving alternate screen and entering it again seems to confuse `notcurses_stop` into not exiting kitty's keyboard protocol.

**Notcurses Version**: 3.0.2

**Terminal**: Kitty 0.23.1

**Output of** `export | egrep 'LANG|LC_CTYPE|TERM'`:
```
COLORTERM=truecolor
LANG=en_US.UTF-8
TERM=xterm-kitty
TERMINAL=kitty
TERMINFO=/usr/lib/kitty/terminfo
```

**Code for reproducing problem:**
```c
#include <stdlib.h>
#include <locale.h>

#include <notcurses/notcurses.h>

int main(void) {
	if (setlocale(LC_ALL, "") == NULL) {
		return EXIT_FAILURE;
	}

	struct notcurses *nc = notcurses_init(NULL, NULL);
	if (nc == NULL) {
		return EXIT_FAILURE;
	}

	notcurses_leave_alternate_screen(nc);
	notcurses_enter_alternate_screen(nc); /* Problem does not occur if we don't enter alternate screen again */

	/* notcurses_stop does not error */
	if (notcurses_stop(nc) < 0) {
		return EXIT_FAILURE;
	}

	return EXIT_SUCCESS;
}
```

**Comment by dankamongmen** [comment]
looking into this today; thanks for the report!

**Comment by dankamongmen** [comment]
problem confirmed (thanks a lot for supplying test code!). i'll have this fixed shortly.

**Comment by dankamongmen** [comment]
so let's see, what's going on here? assuming `NCOPTION_DRAIN_INPUT` is not provided...

1) `notcurses_init()` enters the alternate screen by SMCUP followed by KKEYBOARDENTER. so we're in the kitty protocol in the alternate screen. this all goes to the tty.
2) `leave_alternate_screen()` pops the keyboard setting, then sends RMCUP. we're out of the alternate screen, and the kitty keyboard protocol ought be disabled.

indeed, if we ctrl+c at this point, we do *not* see the problem.

wait....we then reenter the kitty protocol following the rmcup. yet, it doesn't seem that we are?

wait, we enter the protocol with KKBDSUPPORT, but we're passing KKBDENTER...ahhh, cool, KKBDENTER is a PUSH followed by a SET(11). so it ought be *enabled* at the end of `leave_alternate_screen()`, just for the regular screen. interesting.

so we had a push, a pop, and another push, so on our stack ought just be regular keyboard state, and we ought return there with a pop.

3) `enter_alternate_screen()` comes in, *does not* clear the regular screen's keyboard state (this is probably a bug in and of itself), sends SMCUP, then sends a PUSH and a SET(11). that means the stack is now: {regular, 11}, which is almost certainly our problem.

4) `notcurses_stop()` comes along and POPs the stack prior to the RMCUP, doing so whether in the alternate or regular screen. according to (3) above, this will indeed only take us back to SET(11), leading to exactly the problem seen here.

so i think this is all due to failing to POP in `enter_alternate_screen()`. doing so would change the state of the regular screen back to the normal state (as desired), and we would then PUSH the regular state, and see it POPped in `notcurses_stop_minimal()`. yeah, that ought work. we just need ensure we're not calling `enter_alternate_screen()` in general initialization, which i think we are...let's check.

hrmmm, it actually doesn't look like we are?

alright, let's try this, then!

**Comment by dankamongmen** [comment]
yep, that got it =]

**Comment by dankamongmen** [comment]
fix committed in f13346242, will be present in notcurses 3.0.3. thanks a lot for the high-quality bug report! made this much easier.
